packages/grid/_modules_/grid/components/GridCell.tsx

```diff
@@ -68,6 +135,12 @@ export const GridCell: React.FC<GridCellProps> = React.memo((props) => {
         maxHeight: height,
       }}
       tabIndex={tabIndex}
+      onClick={handleClick}
+      onDoubleClick={handleDoubleClick}
+      onMouseEnter={handleEnter}
+      onMouseLeave={handleLeave}
+      onMouseOver={handleHover}
+      onMouseOut={handleOut}
```

Comment:
I wonder if listening to the grid viewport wouldn't help with performance. The changes lead to the creation of six new JavaScript handlers per cell. We can have many cells. I don't benchmark so I have no idea if it's a bottleneck or not. It might be very well marginal.

From a developer perspective, the only difference would be what `event.currentTarget` returns.

Reply:
Actually, we might want to look into adding performance regression tests. The data grid and date picker seem to be two great candidates for it.

Reply:
Well yes originally they were on the root element to optimize the perf.
We can add perf tests but it's out of the scope of this PR I think

Reply:
Performance tests are way outside of the scope of this PR 😁

Reply:
I don't think you need a CI task for a particular issue. (react) profiler devtools should help you sufficiently for specific problems.